tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -469,35 +475,35 @@ class SPIRVEmitter : public ASTConsumer {
       // case <literal_integer>: {...; break;}
       const Expr *caseExpr = caseStmt->getLHS();
       assert(caseExpr);
-      const auto *castExpr = dyn_cast<CastExpr>(caseExpr);
-      assert(castExpr);
-      assert(castExpr->getCastKind() == CastKind::CK_IntegralCast);
-      llvm::APSInt intValue;
-      assert(castExpr->EvaluateAsInt(intValue, astContext,
-                                     Expr::SE_NoSideEffects));
-      const int64_t value = intValue.getSExtValue();
+      assert(caseExpr->isEvaluatable(astContext));
+      Expr::EvalResult evalResult;
+      caseExpr->EvaluateAsRValue(evalResult, astContext);
```

Comment:
actually, `isEvaluatable()` returns False if the expression HasSideEffects. So `assert(caseExpr->isEvaluatable(astContext));` is sufficient.

I will add the code to emit error if the bit width is not 32.